java/dataset/src/test/java/org/apache/arrow/dataset/file/TestFileSystemDataset.java

```diff
@@ -386,6 +388,27 @@ public void testBaseCsvRead() throws Exception {
     }
   }
 
+  @Test
+  public void testResourceToLoad() throws IOException {
+    if (System.getProperty("os.name").equalsIgnoreCase("Windows")) {
+      assertEquals("\\", File.separator);
+      final String pathWithoutFileSeparator = "/" + "avroschema" + "/" + "user.avsc";
+      try (final InputStream stream = TestFileSystemDataset.class.getResourceAsStream(pathWithoutFileSeparator)) {
+        if (stream == null) {
+          throw new FileNotFoundException(pathWithoutFileSeparator);
+        }
+      }
+      final String pathWithFileSeparator = File.separator + "avroschema" + File.separator + "user.avsc";
+      Assertions.assertThrows(FileNotFoundException.class, () -> {
+        try (final InputStream stream = TestFileSystemDataset.class.getResourceAsStream(pathWithFileSeparator)) {
+          if (stream == null) {
+            throw new FileNotFoundException(pathWithFileSeparator);
+          }
+        }
+      }, "FileNotFound \\avroschema\\user.avsc");
+    }
+  }
+
```

Comment:
I don't see why we're testing the Java standard library here. My previous review was asking why this issue was not caught by any test at all on Windows: it implies that we don't test any of this JNI code on Windows, and that is the real issue I want to fix.

Reply:
The current JUnit tests are able to read resources without any issues.

Basically, this error was not detected because of the lack of checking of `getResourceAsStream` in the main code to load native libraries.

Reply:
`File.separator` cause the problem on Windows OS by [ClassLoader](https://docs.oracle.com/javase/8/docs/api/java/lang/ClassLoader.html#getResource-java.lang.String-:~:text=The%20name%20of%20a%20resource%20is%20a%20%27/%27%2Dseparated%20path%20name%20that%20identifies%20the%20resource.) java docs.

Reply:
I don't see how this test changes that. And I don't see what you mean: it seems this should have failed and been caught if any of the tests ran on Windows at all. So the main problem is to make sure we test JNI code on Windows somewhere.

Reply:
Sure, though without that, it's hard to validate that we've fixed things here

Reply:
This test is still not useful